### PR TITLE
Used separate thread for events and perf.

### PR DESCRIPTION
**Pull request by BroMarduk**
OK, I made a real rookie mistake by editing my branch that had the PR the first time I made it not realizing that the PR would take the latest..  I reverted it to working code while I try a few things with making it better IN ANOTHER BRANCH (lesson learned).  

If you decide NOT to implement this PR, please revert my last PR as well as it has the newer, but not working code.

This approach does solve the Signal issues that were occurring causing the thread not to quit all the time.  I've had this issue before with threads and SIGINTs.  Since my project using this code checks for a running instance, the version using the flag variable would error on the previous self.stop() code in the def __del__ method and the thread would not exit (still waiting in the EvDev read_loop).  I believe this occurred due to the fact that some of the variables were already gone at the point the self.stop() code was called.

I did some performance testing with the two methods and found them negligible, but happy to go over the changes with you.

### CODE REVIEW

**Comment by BroMarduk** [comment]
FYI, here is a link to the evdev repository explaining the approach;

https://github.com/gvalkov/python-evdev/issues/15

**Comment by BroMarduk** [comment]
OK, I got this working will using an additional thread for the device read_loop that works better than my original approach.   I also refactored my original commit so that it was backwards compatible with anyone using this library before. 

I'm using this version (well modified slightly for Python 2.7 (queue is Queue) and it has been working great to help replace the missing capacitive touch driver in the latest Raspian.

**Comment by PrzemoF** [comment]
I reverted your first PR, so this one now show a conflict. Can you fix it? Also if you want to support python2 maybe we can use try/except to load Queue/queue? Like here:  https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection

**Comment by BroMarduk** [comment]
Yup. I will fix.  Thanks for the pointer.  I'll make the adjustment so its 2.7 & 3.0 compatible with the try/catch method.

**Comment by BroMarduk** [comment]
OK, the merge was completed and I changed the script to support 2.7+, which required changing the Shebang! at the top.  I don't think this is an issue as I doubt anyone would run this script directly from command line...

**Comment by PrzemoF** [comment]
OK, thanks! I see some more minor problems and one slightly bigger. The minor ones are things like commit title like "merge master into master", whitespace characters in empty lines and line lengths. Also the empty line at the end is not required - there has to be newline character, not a whole new line. The big one is that this won't work in python3:

    except Exception, ex:

I use vim/flake8, so I'll have to fix them just to keep the error window clean. Are you OK if I take your code, split it into atomic commits where possible, fix the minor problems and then commit it to the main branch?

That's what I got from flake8 check:

```
$ flake8 pitft_touchscreen.py
pitft_touchscreen.py:12:1: W293 blank line contains whitespace
pitft_touchscreen.py:13:1: W293 blank line contains whitespace
pitft_touchscreen.py:32:80: E501 line too long (125 > 79 characters)
pitft_touchscreen.py:35:25: E999 SyntaxError: invalid syntax
pitft_touchscreen.py:36:80: E501 line too long (99 > 79 characters)
pitft_touchscreen.py:42:8: E114 indentation is not a multiple of four (comment)
pitft_touchscreen.py:42:39: W291 trailing whitespace
pitft_touchscreen.py:76:1: W293 blank line contains whitespace
pitft_touchscreen.py:86:1: W293 blank line contains whitespace
pitft_touchscreen.py:89:1: W293 blank line contains whitespace
pitft_touchscreen.py:92:1: W391 blank line at end of file
pitft_touchscreen.py:92:1: W293 blank line contains whitespace

P.S. I should have time to test your code on the real hardware soon

**Comment by BroMarduk** [comment]
Sure.  Feel free to modify as you see fit or to take what you want from this PR and "atomize" into better commits as you see fit.  And thanks for the heads-up on Flake8.   In the meantime I fixed the Flake8 warnings (no longer reported) and added a second example that you can try out.   If you run the example and kill the process or shutdown the system, the threads in the new main class exit cleanly as expected.  There is a case where evdev will throw an exception on the read when closing down and mermory has been cleaned up, but I'm not planning on getting into that code.   

I added the sample so you could test.   Feel free to include it or not!

**Comment by PrzemoF** [comment]
Can you check if you're OK with: https://github.com/PrzemoF/pitft_touchscreen/commits/pr3

**Comment by BroMarduk** [comment]
I tested out the pr3 branch and all looks good.   I do think I found an issue (potentially) with the touch screen code.   I was doing some device testing and found that when clicking repetitively on or around the same spot on the TFT would cause events at random spots where the touch was not.  I can show this by showing the output of the example:

**Comment by BroMarduk** [comment]
tl;dr;  pitft_touchscreen seems to be processing events correctly from evdev.  Nothing seems wrong.

OK, the problem I have is that the messages coming from EV occasion a ABS_MT_TRACKING_ID comes in with only an X or Y ABS value from my device.   This results in a message in the queue with a X or Y value of None.   I don't think this is an issue with the event generator here as this is actually what is happening from evdev.   My code processing the event queue just needs to change.    Here is a sample evtest output - strangeness (to me) highlighted in bold.

Event: time 1546916078.486824, -------------- SYN_REPORT ------------
Event: time 1546916078.642429, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value 3108
Event: time 1546916078.642429, type 3 (EV_ABS), code 53 (ABS_MT_POSITION_X), value 92
Event: time 1546916078.642429, type 3 (EV_ABS), code 54 (ABS_MT_POSITION_Y), value 93
Event: time 1546916078.642429, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 1
Event: time 1546916078.642429, type 3 (EV_ABS), code 0 (ABS_X), value 92
Event: time 1546916078.642429, type 3 (EV_ABS), code 1 (ABS_Y), value 93
Event: time 1546916078.642429, -------------- SYN_REPORT ------------
Event: time 1546916078.691553, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value -1
Event: time 1546916078.691553, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 0
Event: time 1546916078.691553, -------------- SYN_REPORT ------------
Event: time 1546916078.834876, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value 3109
Event: time 1546916078.834876, type 3 (EV_ABS), code 53 (ABS_MT_POSITION_X), value 90
Event: time 1546916078.834876, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 1
**Event: time 1546916078.834876, type 3 (EV_ABS), code 0 (ABS_X), value 90
Event: time 1546916078.834876, -------------- SYN_REPORT ------------
Event: time 1546916078.908329, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value -1**    
Event: time 1546916078.908329, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 0
Event: time 1546916078.908329, -------------- SYN_REPORT ------------
Event: time 1546916079.086668, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value 3110
Event: time 1546916079.086668, type 3 (EV_ABS), code 54 (ABS_MT_POSITION_Y), value 96
Event: time 1546916079.086668, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 1
**Event: time 1546916079.086668, type 3 (EV_ABS), code 1 (ABS_Y), value 96
Event: time 1546916079.086668, -------------- SYN_REPORT ------------
Event: time 1546916079.145789, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value -1**
Event: time 1546916079.145789, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 0
Event: time 1546916079.145789, -------------- SYN_REPORT ------------
Event: time 1546916079.328069, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value 3111
Event: time 1546916079.328069, type 3 (EV_ABS), code 53 (ABS_MT_POSITION_X), value 88
Event: time 1546916079.328069, type 3 (EV_ABS), code 54 (ABS_MT_POSITION_Y), value 97
Event: time 1546916079.328069, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 1
Event: time 1546916079.328069, type 3 (EV_ABS), code 0 (ABS_X), value 88
Event: time 1546916079.328069, type 3 (EV_ABS), code 1 (ABS_Y), value 97
Event: time 1546916079.328069, -------------- SYN_REPORT ------------
Event: time 1546916079.398698, type 3 (EV_ABS), code 57 (ABS_MT_TRACKING_ID), value -1
Event: time 1546916079.398698, type 1 (EV_KEY), code 330 (BTN_TOUCH), value 0
Event: time 1546916079.398698, -------------- SYN_REPORT ------------

**Comment by PrzemoF** [comment]
I just pushed your changes from the pr2 branch, so I'm going to close this PR . Please open an issue if you see some other strange behaviour - we can discuss it there.

Thanks again!
